src/ReadStream.h

```diff
@@ -59,19 +64,132 @@ class ReadStream {
     virtual ~ReadStream() = default;
 
 protected:
+    /**
+     * Read a record from a string.
+     *
+     * @param source - string containing a record
+     * @param recordTypeName - record type.
+     * @parem pos - start parsing from this position.
+     * @param consumed - if not nullptr: number of characters read.
+     *
+     */
+    RamDomain readRecord(const std::string& source, const std::string& recordTypeName, size_t pos = 0,
+            size_t* _consumed = nullptr) {
```

Comment:
`consumed`, or `charactersRead`, rather than `_consumed`

That aside, could it be a member variable instead? That has its own ugliness, but would avoid the need to pass the pointer around in the parameters. This is not a request for a change as I'm not sure it's an improvement.